src/dropdown/dropdown.ts

```diff
@@ -82,7 +89,8 @@ export class NgbDropdown {
   }
 
   closeFromOutsideClick($event) {
-    if (this.autoClose && $event.button !== 2 && !this._isEventFromToggle($event)) {
+    if (this.autoClose && this._shouldcloseFromInsideClick($event) && $event.button !== 2 &&
```

Comment:
I don't understand this logic... Why is the `_shouldcloseFromInsideClick` impacting logic of outside clicks?